src/engine/sched.c

```diff
@@ -1882,7 +1882,7 @@ watchdog_enabled(struct dss_xstream *dx)
 	if (sched_unit_runtime_max == 0)
 		return false;
 
-	return dx->dx_xs_id == 0 || (sched_watchdog_all && dx->dx_main_xs);
+	return dx->dx_xs_id == 0 || (sched_watchdog_all && dss_xstream_has_context(dx));
```

Comment:
I don't think we should change anything related to watchdog in this PR.

Reply:
OK